src/sota_tools/ostree_object.cc

```diff
@@ -228,19 +236,22 @@ void OSTreeObject::Upload(const TreehubServer &push_target, CURLM *curl_multi_ha
   curlEasySetoptWrapper(curl_handle_, CURLOPT_WRITEDATA, this);
   http_response_.str("");  // Empty the response buffer
 
-  assert(form_post_ == nullptr);
-  struct curl_httppost *last_item = nullptr;
-  const CURLFORMcode form_err =
-      curl_formadd(&form_post_, &last_item, CURLFORM_COPYNAME, "file", CURLFORM_FILE, file_path_.c_str(), CURLFORM_END);
-  if (form_err != 0u) {
-    // Apparently there is not strerror for formadd.
-    LOG_ERROR << "curl_formadd error: " << form_err;
+  struct stat file_info {};
+  fd_ = fopen(file_path_.c_str(), "rb");
+  if (fd_ == nullptr) {
+    LOG_ERROR << "Could not open file: " << file_path_.c_str();
+  } else {
+    if (stat(file_path_.c_str(), &file_info) < 0) {
+      LOG_ERROR << "Could not get file information";
+    }
   }
+  curlEasySetoptWrapper(curl_handle_, CURLOPT_READDATA, fd_);
+  curlEasySetoptWrapper(curl_handle_, CURLOPT_POSTFIELDSIZE, file_info.st_size);
   curlEasySetoptWrapper(curl_handle_, CURLOPT_POST, 1);
-  curlEasySetoptWrapper(curl_handle_, CURLOPT_HTTPPOST, form_post_);
 
-  curlEasySetoptWrapper(curl_handle_, CURLOPT_PRIVATE, this);  // Used by ostree_object_from_curl
```

Comment:
Is it correct to remove this line? It may be, but it seems like it is still used.

Reply:
Didn't remove the line. See line 254.

Reply:
Thanks, don't know how I missed that.